### DISCUSSION TITLE
Deleting entity from edit dialog without confirmation

### DISCUSSION THREAD

**Discussion by ghost** [serenity-is/Serenity#5909] [Help]
Hello,

Clicking on "Delete" button inside row's edit dialog displays a confirmation dialog. How to disable confirmation dialog to be able to delete row on one click?

thank you

**Comment by reach2rv** [comment]
protected getToolbarButtons(): Serenity.ToolButton[] {
            let btns = super.getToolbarButtons();
            var btnDelete = Q.first(btns, x => x.cssClass == "delete-button");
            var oldDeleteClick = btnDelete.onClick;
            btnDelete.onClick = e => { this.confirmBeforeDelete(oldDeleteClick, e); }
 
            return btns;
        }
        
        // **** Validate Reference of Voucher in Stock Transaction Before Delete. ****
        private confirmBeforeDelete(oldEvt, e) {
            let self = this;
           ApprovalInService.Delete(
                            {
                                EntityId: this.entityId
                            }, response => {
                                Serenity.SubDialogHelper.triggerDataChange(self);
                                self.dialogClose();
                                window.setTimeout(() => Q.notifySuccess(Q.text('Deleted successfully')), 0);
                                    
                            }
                        );
                    }
                })
            }

**Reply by reach2rv** [reply]
You may get some errors as I have pasted from mobile
